mp/src/game/shared/momentum/mom_stickybomb.cpp

```diff
@@ -0,0 +1,357 @@
+
+        rotInterpolator.AddToHead(changeTime, &vCurAngles, false);
+    }
+}
+
+int CMomStickybomb::DrawModel(int flags)
+{
+    // Don't draw the stickybomb during the time defined by the cvar.
+    if (gpGlobals->curtime - m_flSpawnTime < mom_sj_stickybomb_drawdelay.GetFloat())
+    {
+        return 0;
+    }
+
+    return BaseClass::DrawModel(flags);
+}
+
+void CMomStickybomb::Simulate()
+{
+    if (!m_bPulsed)
+    {
+        if ((gpGlobals->curtime - m_flCreationTime) >= 0.8f)
+        {
+            ParticleProp()->Create("stickybomb_pulse_red", PATTACH_ABSORIGIN_FOLLOW);
+            m_bPulsed = true;
+        }
+    }
+    BaseClass::Simulate();
+}
+
+#else
+
+void CMomStickybomb::SetupInitialTransmittedGrenadeVelocity(const Vector &velocity) { m_vInitialVelocity = velocity; }
+
+CMomStickybomb *CMomStickybomb::Create(const Vector &position, const QAngle &angles, const Vector &velocity,
+                                       const AngularImpulse &angVelocity, CBaseCombatCharacter *pOwner)
+{
+    CMomStickybomb *pStickybomb =
+        static_cast<CMomStickybomb *>(CBaseEntity::CreateNoSpawn("momentum_stickybomb", position, angles, pOwner));
+
+    if (pStickybomb)
+    {
+        DispatchSpawn(pStickybomb);
+        pStickybomb->InitStickybomb(velocity, angVelocity, pOwner);
+        pStickybomb->ApplyLocalAngularVelocityImpulse(angVelocity);
+        pStickybomb->SetAbsVelocity(velocity);
+        pStickybomb->SetThrower(pOwner);
+        pStickybomb->SetAbsAngles(angles);
+
+        pStickybomb->SetDamage(120.0f);
+        pStickybomb->SetRadius(146.0f);
+    }
+
+    return pStickybomb;
+}
+
+void CMomStickybomb::InitStickybomb(const Vector &velocity, const AngularImpulse &angVelocity,
+                                    CBaseCombatCharacter *pOwner)
+{
+    SetupInitialTransmittedGrenadeVelocity(velocity);
+
+    IPhysicsObject *pPhysicsObject = VPhysicsGetObject();
+    if (pPhysicsObject)
+    {
+        pPhysicsObject->AddVelocity(&velocity, &angVelocity);
+    }
+}
+
+void CMomStickybomb::RemoveStickybomb(bool bNoGrenadeZone)
+{
+    // Kill it
+    SetThink(&BaseClass::SUB_Remove);
+    SetNextThink(gpGlobals->curtime);
+    SetTouch(nullptr);
+    AddEffects(EF_NODRAW);
+
+    if (bNoGrenadeZone)
+    {
+        // Sprite flash
+        CSprite *pGlowSprite = CSprite::SpriteCreate(NOGRENADE_SPRITE, GetAbsOrigin(), false);
+        if (pGlowSprite)
+        {
+            pGlowSprite->SetTransparency(kRenderGlow, 255, 255, 255, 255, kRenderFxFadeFast);
+            pGlowSprite->SetThink(&CSprite::SUB_Remove);
+            pGlowSprite->SetNextThink(gpGlobals->curtime + 1.0);
+        }
+    }
+}
+
+void CMomStickybomb::Detonate()
+{
+    trace_t tr;
+    Vector vecSpot;
+
+    SetThink(nullptr);
+
+    vecSpot = GetAbsOrigin() + Vector(0, 0, 8);
+    UTIL_TraceLine(vecSpot, vecSpot + Vector(0, 0, -32), MASK_SHOT_HULL, this, COLLISION_GROUP_NONE, &tr);
+
+    Explode(&tr, tr.m_pEnt);
+
+    if (m_bFizzle)
```

Comment:
Should this block move to Explode, right under the check for if it's in a no grenade zone? If this logic is correct, this will explode the sticky and then fizzle it, doesn't fizzling mean to not explode?